submissions/oluwaboriShobayo/DOM/signUp.js

```diff
@@ -0,0 +1,95 @@
+let form = document.getElementById("sign");
+     firstname = document.getElementById("firstname");
+     lastname = document.getElementById("lastname");
+     othername = document.getElementById("othername");
+     phonenumber = document.getElementById("phonenumber");
+     mail  = document.getElementById("mail");
+     password = document.getElementById("password");
+     password2 = document.getElementById("password2");
+     nameRegex = /^[A-Za-z]+$/;
+     numRegex = /^\(?([0-9]{3})\)?[-. ]?([0-9]{3} )[-. ]?([0-9]{4})$/;
+     emailRegex = /^([\w-]+(?:\.[\w-]+)*)@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$/;
+     pwdRegex = /^(?=.*?[A-Z])(?=(.*[a-z]){1,})(?=(.*[\d]){1,})(?=(.*[\W]){1,})(?!.*\s).{8,}$/; 
+
+// // firstname VALIDATION
+function emptyField (event) {
+    if (firstname.value == '') {
+        firstname.nextElementSibling.innerHTML = "Please enter your firstname";
+        event.preventDefault();
+}
```

Comment:
You are still repeating one thing over and over again. Why don't you write a single function called `emptyField` that take `fieldName` and `errorMessage` as a parameter i.e `emptyField(fieldName, errorMessage, event)`. Then the function will hold this block of code as seen below 
```
if (fieldName.value == '') {
        fieldName.nextElementSibling.innerHTML = errorMessage;
        event.preventDefault();
}
```
Then when calling the function, you can then pass in the argument. E,g for firstname, you can call the function like this `emptyField(firstname, "Please enter your first name")`


Do a similar approach for the regex check